docs/src/main/asciidoc/central-logging.adoc

```diff
@@ -0,0 +1,369 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Central log management (Graylog, Logstash, Fluentd)
+
+include::./attributes.adoc[]
+
+This guide explains how you can send you log to a central logging system like Graylog, Logstash (inside the Elastic Stack or ELK - Elasticsearch, Logstash, Kibana) or
+Fluentd (inside EFK - Elasticsearch, Fluentd, Kibana).
+
+There is a lot of different way to centralize your logs (if you're using Kubernetes, the simplest way is to log to the console and ask you cluster administrator to integrate a central log manager inside your cluster).
+In this guide, we will expose how to send them to an external tool using the `quarkus-logging-gelf` extension that can use TCP or UDP to send logs in the Graylog Extented Log Format (GELF).
+
+The `quarkus-logging-gelf` extension will add a GELF log handler to the underlying logging backend that Quarkus uses (jboss-logmanager).
+By default, it is disabled, if you enable it but still use another handler (by default the console handler is enabled), your logs will be sent to both handlers.
+
+== Example application
+
+The following examples will all be based on the same example application that you can create based on the following steps:
+
+- Create an application with the `quarkus-logging-gelf` extension. You can use the following Maven command to create it:
+```
+mvn io.quarkus:quarkus-maven-plugin:1.0.0.Final:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=gelf-logging \
+    -DclassName="org.acme.quickstart.GelfLoggingResource" \
+    -Dpath="/gelf-logging" \
+    -Dextensions="logging-gelf"
+```
+
+- For demonstration purpose, we create an endpoint that do nothing but log a sentence. You don't need to do this inside your application.
+```
+@Path("/gelf-logging")
+@ApplicationScoped
+public class GelfLoggingResource {
+    private static final Logger LOG = Logger.getLogger(GelfLogHandlerResource.class);
+
+    @GET
+    public void log() {
+        LOG.info("Some useful log message");
+    }
+
+}
+```
+
+- Configure the gelf log handler to send logs to an external UDP endpoint on the port 12201
+```
+quarkus.log.handler.gelf.enabled=true
+quarkus.log.handler.gelf.host=localhost
+quarkus.log.handler.gelf.port=12201
+```
+
+== Send logs to Graylog
+
+To send logs to Graylog, you first need to launch the components that compose the Graylog stack:
+
+- MongoDB
+- Elasticsearch
+- Graylog
+
+You can do this via the following docker-compose file that you can launch via `docker-compose run -d`:
+
+```
+version: '3.2'
+
+services:
+  elasticsearch:
+    image: docker.elastic.co/elasticsearch/elasticsearch-oss:6.8.2
```

Comment:
We can't use a more recent version?

Reply:
This is the version used inside the Graylog documentation: http://docs.graylog.org/en/3.1/pages/installation/docker.html#here
A more recent version should works but I will need to re-run all the test ...

Reply:
In fact Graylog is only compatible with Elasticsearch 5.6 and 6.8: https://docs.graylog.org/en/3.1/pages/configuration/elasticsearch.html

The other examples should work on more recent version of Elasticsearch but to keep it simple I prefere to use the same for all